.rancher-pipeline.yml

```diff
@@ -0,0 +1,28 @@
+stages:
+  - name: Build
+    steps:
+      - runScriptConfig:
+          image: maven:3-openjdk-8
+          shellScript: mvn clean install -DskipTests
+  - name: Build Docker
+    steps:
+      - publishImageConfig:
+          dockerfilePath: ./Dockerfile
+          buildContext: .
+          tag: docker.dev.folio.org/mod-circulation:core-functional-latest
+          pushRemote: true
+          registry: docker.dev.folio.org
+  - name: Deploy
+    steps:
+      - applyAppConfig:
+          catalogTemplate: p-9mwdx:core-functional-helmcharts-mod-circulation
+          version: 0.1.18
+          answers:
+            image.repository: docker.dev.folio.org/mod-circulation
+            image.tag: core-functional-latest
+            resources.limits.cpu: 1000m
+            resources.limits.memory: 3000Mi
+          name: mod-circulation
+          targetNamespace: core-functional
+timeout: 60
+notification: {}
```

Comment:
Should this be merged to master?

Reply:
I think yes, it is pipeline configuration.

Reply:
My understanding is that the dev scratch environments are intended for testing non-mainline versions of the modules and that they have their own configuration for deploying the mainline versions.

Why would we want the mainline to have this configuration? Would any master builds overwrite the docker image being used by the scratch environments?

It is my understanding that we would only want a single branch with this configuration, otherwise the docker image used might oscillate between versions produced by different branches.

I think it is worth asking the folks who set up the environment if this is recommended practice.

Reply:
> My understanding is that the dev scratch environments are intended for testing non-mainline versions of the modules and that they have their own configuration for deploying the mainline versions.

That's true, however we can not run pipeline without this file committed. 

> Why would we want the mainline to have this configuration? 

Because it defines the pipeline flow, the only variable here is image.tag which can be overridden if needed. 

> It is my understanding that we would only want a single branch with this configuration, otherwise the docker image used might oscillate between versions produced by different branches.

It is true, but the pipeline won't be started by some automated process, only when we explicitly run the pipeline for a branch on runcher.

Reply:
> That's true, however we can not run pipeline without this file committed.

As I understand it, we would only want to do that on temporary branches. Is that not the case?

I think it is worthwhile asking the folks who built the rancher tooling (or the pilot team, FoliJet) what their guidance is for this.

If we are to put this file on the mainline, I would prefer it be located in a separate directory and named as a template, so it cannot be accidentally used on an incorrect branch.

What do you think to that idea?

Reply:
I'm not sure if rancher can pick it from the directory. I'd remove it at all then, because it does not bring any value. My initial idea was that any feature branch can be quickly built and deployed without any additional actions needed. 

Hi @TarasSpashchenko, do you have any guidance for this? Is it fine to have this pipeline file in the master branch?

Reply:
> I'm not sure if rancher can pick it from the directory. I'd remove it at all then, because it does not bring any value. My initial idea was that any feature branch can be quickly built and deployed without any additional actions needed.

That was my intention with a template being kept in a separate directory. The file would be easy to refer to when creating a branch for the new environments yet not be accidentally picked up if not intended.

If we want any branch to be usable with this approach then I agree with your approach. I think we need to make sure that intent is very clear and understood by folks.

I think it could be confusing for folks using the repository to find contradictory information about builds and Docker repositories.

Reply:
Hi Marc, I asked EPAM folks that did set-up of rancher, and they clarified following:

> Yes, this file should be kept in master and then, we can change (in most cases means override image tag) the file in our feature branch, if needed.

Reply:
Ok. I guess I need to accept this as common practice then. Is that viewpoint written up somewhere?

Please can this be documented in the readme, at least so folks might be aware that this configuration is specific to these development environments and that the image tag should be overridden.

I anticipate that folks will forget to do so and accidentally corrupt other folks environments :-/

Reply:
I've added a readme for this.